src/main/scala/scorex/core/core.scala

```diff
@@ -20,16 +21,29 @@ package object core {
   type VersionTag = VersionTag.Type
 
   def idsToString(ids: Seq[(ModifierTypeId, ModifierId)])
-                 (implicit encoder: BytesEncoder): String = (ids.headOption, ids.lastOption) match {
-    case (Some(f), Some(l)) if f._2 sameElements l._2 => s"[(${f._1},${encoder.encode(f._2)})]"
+                 (implicit encoder: ScorexEncoder): String = (ids.headOption, ids.lastOption) match {
+    case (Some(f), Some(l)) if f._2 == l._2 => s"[(${f._1},${encoder.encode(f._2)})]"
     case (Some(f), Some(l)) => s"[(${f._1},${encoder.encode(f._2)})..(${l._1},${encoder.encode(l._2)})]"
     case _ => "[]"
   }
```

Comment:
This could be replaced by:
```
  def idsToString(ids: Seq[(ModifierTypeId, ModifierId)])(implicit enc: ScorexEncoder): String = {
    List(ids.headOption, ids.lastOption)
      .flatten
      .map { case (typeId, id) => s"($typeId,${enc.encode(id)})" }
      .mkString("[", "..", "]")
  }
```

Reply:
fixed